typescript/cli/src/deploy/warp.ts

```diff
@@ -568,63 +568,67 @@ function splitWarpApplyConfig(
   );
 }
 
-export async function extendWarpRoute(
+/**
+ * Handles the deployment and configuration of new contracts for extending a Warp route.
+ * This function performs several key steps:
+ * 1. Derives metadata from existing contracts and applies it to new configurations
+ * 2. Deploys new contracts using the derived configurations
+ * 3. Merges existing and new router configurations
+ * 4. Generates an updated Warp core configuration
+ */
+async function deployWarpExtensionContracts(
   params: WarpApplyParams,
   apiKeys: ChainMap<string>,
+  existingConfigs: WarpRouteDeployConfig,
+  initialExtendedConfigs: WarpRouteDeployConfig,
   warpCoreConfigByChain: ChainMap<WarpCoreConfig['tokens'][number]>,
-): Promise<{
-  updatedWarpDeployConfig: WarpRouteDeployConfig;
-  updatedWarpCoreConfigByChain: ChainMap<WarpCoreConfig['tokens'][number]>;
-}> {
-  const { context, warpDeployConfig } = params;
-  const warpCoreChains = Object.keys(warpCoreConfigByChain);
-
-  // Split between the existing and additional config
-  const [existingConfigs, initialExtendedConfigs] = splitWarpApplyConfig(
-    warpDeployConfig,
-    warpCoreChains,
-  );
-
-  const extendedChains = Object.keys(initialExtendedConfigs);
-  if (extendedChains.length === 0) {
-    return {
-      updatedWarpDeployConfig: { ...warpDeployConfig },
-      updatedWarpCoreConfigByChain: { ...warpCoreConfigByChain },
-    };
-  }
-
-  logBlue(`Extending Warp Route to ${extendedChains.join(', ')}`);
-
+) {
+  // Deploy new contracts with derived metadata
   const extendedConfigs = await deriveMetadataFromExisting(
-    context.multiProvider,
+    params.context.multiProvider,
     existingConfigs,
     initialExtendedConfigs,
   );
 
-  // Deploy new contracts
   const newDeployedContracts = await executeDeploy(
     {
-      context,
+      context: params.context,
       warpDeployConfig: extendedConfigs,
     },
     apiKeys,
   );
 
   // Merge existing and new routers
   const mergedRouters = mergeAllRouters(
-    context.multiProvider,
+    params.context.multiProvider,
     existingConfigs,
     newDeployedContracts,
     warpCoreConfigByChain,
   );
 
-  // Get the updated core config that includes both existing and new contracts
+  // Get the updated core config
   const { warpCoreConfig: updatedWarpCoreConfig, addWarpRouteOptions } =
     await getWarpCoreConfig(params, mergedRouters);
   WarpCoreConfigSchema.parse(updatedWarpCoreConfig);
 
-  // Create new warpDeployConfig with updated addresses
-  const updatedWarpDeployConfig = {
+  return {
+    newDeployedContracts,
+    updatedWarpCoreConfig,
+    addWarpRouteOptions,
+  };
+}
+
+/**
+ * Updates the Warp route configuration by incorporating newly deployed contract addresses.
```

Comment:
Let's be specific in which configuration, i.e. in this case `Warp Deploy Config`